ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitter.scala

```diff
@@ -448,30 +447,31 @@ private[kvutils] class TransactionCommitter(
       blindingInfo: BlindingInfo,
       commitContext: CommitContext,
   ): Unit = {
-    val effects = InputsAndEffects.computeEffects(transactionEntry.transaction)
-    val cid2nid: Value.ContractId => NodeId =
-      transactionEntry.transaction.localContracts
+    val localContracts = transactionEntry.transaction.localContracts
+    val consumedContracts = transactionEntry.transaction.consumedContracts
+    val contractKeys = transactionEntry.transaction.updatedContractKeys
     // Add contract state entries to mark contract activeness (checked by 'validateModelConformance').
-    for ((key, createNode) <- effects.createdContracts) {
+    localContracts.foreach { case (cid, (nid, createNode)) =>
```

Comment:
Why not use for comprehension for better readability?
```suggestion
    for((cid, (nid, createNode)) <- localContracts) {
```

Reply:
Let me expose the original motivation for the change I made here.

- @stephencompall-DA strongly advices to never use the pattern matching in the left-hand side of a `<-` inside a for comprehension as it is not translate as you expect. (It think the reason is explain [here](https://github.com/lampepfl/dotty/issues/2573, @stephencompall-DA please correct me if I mistake). 
If you want to keep using for comprehension, you should rather use 
```
for {
  entry <- effects.createdContracts) 
  ((key, createNode) = entry 
} { ... }
``` 

- I personally prefer avoiding for-comprehension for imperative code.

Reply:
Correct, and I wrote a mini-article about it [on our forum](https://discuss.daml.com/t/pattern-always-adds-withfilter-scala-ritual/1463). And I also agree with your motivation for avoiding the `for` without `yield`.